pyccel/ast/core.py

```diff
@@ -1411,7 +1411,8 @@ class Iterable(PyccelAstNode):
                - n_indices
                - to_range
     """
-    acceptable_iterator_types = (Variable, PythonMap, PythonZip, PythonEnumerate, PythonRange, IndexedElement)
+    acceptable_iterator_types = (Variable, PythonMap, PythonZip, PythonEnumerate,
+                                  PythonRange, PythonTuple, PythonList, IndexedElement)
```

Comment:
Yes, it is used to indicate `_num_indices_required`

Reply:
When do you create an iterator of a `PythonList`?

Reply:
I was wrong sorry, since we are not making for loops when trying to extend a list then it's unused. So far in this PR, `_num_indices_required` is only set in the case of `PythonRange`

Reply:
updated 2930d873b688385ae34644d20b2dacfbafcfcb87